service/ui.go

```diff
@@ -175,15 +175,16 @@ func (uis *UIServer) GetCommonViewData(w http.ResponseWriter, r *http.Request, n
 	projectCtx, err := GetProjectContext(r)
 	if err != nil {
 		grip.Errorf(errors.Wrap(err, "error getting project context").Error())
-		uis.projectNotFoundBase(w, r, nil)
 		return ViewData{}
 	}
 	if needsProject {
 		var project *model.Project
 		project, err = projectCtx.GetProject()
 		if err != nil || project == nil {
+			if err == nil && project == nil {
+				err = errors.New("no project found")
+			}
```

Comment:
Probably, but I'm not going to handle it in this ticket, but this case is also an unlikely case to hit right now. It only got hit because an invalid version document was inserted into the smoke test data recently.

All the places where `GetCommonViewData` is used assume that it will succeed; on error, it will mostly just break the page view (or for the thing I changed on this line, panic). There's no way to write a response in `GetCommonViewData` right now because the UIServer always uses the the resulting data from this to render the actual page it wants to show and it's invalid to call `WriteResponse` multiple times. A better way to do this would be to either 1. actually return the error and handle it at a higher level or 2. write a 404 response from this and return a bool/error indicating that it wrote a response already.

I made [a ticket](https://jira.mongodb.org/browse/EVG-7505) to handle the error case better.